precode.go

```diff
@@ -56,7 +81,16 @@ func main() {
 
 	// 4. Собираем числа из каналов outs
 	// ...
-
+	for i := 0; i < NumOut; i++ {
+		wg.Add(1)
+		go func() {
```

Comment:
Не могу подобрать код, чтобы сломать твой пример, но тут лучше делать захват `i`, так как если это не сделать, то все горутины будут работать с одной переменной, которая меняется в цикле, а захват помгает это избежать
```suggestion
		go func(i int64) {
```